carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/users/ExpectedParticipantData.kt

```diff
@@ -0,0 +1,66 @@
+package dk.cachet.carp.common.application.users
+
+import dk.cachet.carp.common.application.data.input.InputDataType
+import kotlinx.serialization.Serializable
+
+
+/**
+ * Describes a participant [attribute] [assignedTo] to be input by specified users in a study.
+ */
+@Serializable
+data class ExpectedParticipantData(
+    val attribute: ParticipantAttribute,
+    val assignedTo: AssignedTo = AssignedTo.Anyone
+)
+{
+    val inputDataType: InputDataType
+        get() = attribute.inputDataType
+}
+
+
+/**
+ * Determines whether the set contains any of the following conflicts:
+ * - contains differing participant attributes with the same input type
+ * - contains multiple attributes of the same input type which are assigned to the same role
+ *
+ * @throws IllegalArgumentException if [exceptionOnConflict] is set to true and the set contains a conflict.
+ */
+fun Set<ExpectedParticipantData>.hasNoConflicts( exceptionOnConflict: Boolean = false ): Boolean
```

Comment:
Yeah. I had it on my radar. I even think I added it in a third place in the meantime. But, this would be internal refactoring that doesn't impact API, so later.

The advantage of the current approach is since `Set<ExpectedParticpipantData>` is used in serialized types, the serialized format is more sane than yet another class wrapper, without having to introduce some custom serializer. In other words, I agree, but postponed for now since it's not urgent.